Report/sections/3_implementation/error_handling.tex

```diff
@@ -0,0 +1,70 @@
+\section{Error handling}
+It is useful for users to get error messages detailing any syntax errors there might have been found during parsing.
+
+A simple solution is to provide the user with the offending token and/or line. This is very basic error output, which could be augmented by a detailed message telling the user what is wrong; i.e. a missing semicolon. Even better if the message told the user how to fix the error. 
+
+Going a level further, the parser could attempt to make a recovery. Handle the error, however possible it may be. There are a few ways to do this.
+
+First is panic-mode recovery, in which the parser removes input tokens until a synchronizing token is found. In the case of \dazel{}, this could be a semicolon, indicating the end of a statement. This is a simple solution, but it may skip a considerable amount of input, which may produce unexpected results\cite{crafting_a_compiler}.
+
+It is also possible to do phrase-level recovery. This means to perform corrections locally in some erroneous phrase. It could be to insert a missing semicolon, or even delete an extraneous one. In performing corrections, one must be careful not to enter an infinite loop\cite{spo_course}.
```

Comment:
```suggestion
It is also possible to do phrase-level recovery. This means performing corrections locally in some erroneous phrase, for example inserting a missing semicolon or even deleting an extraneous one. In performing corrections, one must be careful not to enter an infinite loop\cite{spo_course}.
```

Reply:
```suggestion
It is also possible to do phrase-level recovery. This means performing corrections locally in some erroneous phrase. This could be something as trivial as inserting a missing semicolon, or even deleting an extraneous one. In performing corrections, one must be careful not to enter an infinite loop\cite{spo_course}.
```